controllers/storageclass/storageclass_controller.go

```diff
@@ -118,50 +118,71 @@ func (r *StorageClassWatcher) Reconcile(_ context.Context, request reconcile.Req
 		return result, err
 	}
 
-	flashSystemCluster, fscErr := r.getFlashSystemClusterByStorageClass(sc)
-	if fscErr == nil {
-		fscName := flashSystemCluster.GetName()
-		r.Log.Info("FlashSystemCluster found", "FlashSystemCluster", fscName)
-
-		// Check GetDeletionTimestamp to determine if the object is under deletion
-		if !sc.GetDeletionTimestamp().IsZero() {
-			r.Log.Info("object is terminated")
-			err = r.removeStorageClassFromConfigMap(*configMap, fscName, request.Name)
-			if err != nil {
-				return result, err
-			}
-		} else {
-			poolName, ok := sc.Parameters[util.CsiIBMBlockScPool]
-			if ok {
-				if err = r.addStorageClassToConfigMap(*configMap, fscName, request.Name, poolName); err != nil {
+	isTopology := false
+	if _, ok := sc.Parameters["by_management_id"]; ok {
+		isTopology = true
+		r.Log.Info("StorageClass is a topology aware StorageClass", "isTopology", isTopology)
```

Comment:
you can short these lines into
`_, isTopology = sc.Parameters["by_management_id"]` since the second parameter is of type boolean and it's indicate whether the key exist.